typescript/src/schema/proto.ts

```diff
@@ -106,11 +132,17 @@ function toInterface(type: string) {
   return type.replace(/\.([^.]+)$/, '.I$1');
 }
 
+function toLRInterface(type: string, inputType: string) {
+  return inputType.substring(1, inputType.lastIndexOf('.') + 1) + 'I' + type;
```

Comment:
This deserves a comment. Even better, I strongly prefer making this a regex replace since it's often much more readable.